### PR TITLE
Add: Added homepage to Video library

### PR DESCRIPTION
**Pull request by zoef-yt**
In this Pr, I added a Homepage to my video library.

### CODE REVIEW

**Review by Muzammil-khan710 — APPROVED** [review]
UI is really great Zoef !
Code structure is good
Good to merge

**Review by iRohitGaur — APPROVED** [review]
Nice work!

Code is easy to understand.

Good to merge 👍🏻

File: src/Pages/Home/Home.css [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+		opacity: 1;
+		z-index: 5;
+		background-color: var(--colour-black-fifth);
+		padding: 1rem;
+	}
+
+	.category-card .btn {
+		position: absolute;
+		bottom: 0rem;
+	}
```
Comment (bhushannaikaroskar) [inline src/Pages/Home/Home.css:115]: Maybe you can use just 0 instead of 0rem.

File: src/Pages/Components/Navbar/Navbar.jsx [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+						<div className='profile-modal'>
+							<li>My Profile</li>
+							<hr />
+							<li>My Playlist</li>
+							<hr />
+							<li>Watch Later</li>
+							<hr />
+							<li>Liked Videos</li>
+							<hr />
+							<li>Logout 😞</li>
```
Comment (bhushannaikaroskar) [inline src/Pages/Components/Navbar/Navbar.jsx:31]: I would suggestion to remove the <hr/> tag. It works the same even without <hr/> tag

**Review by bhushannaikaroskar — APPROVED** [review]
Hey zoef, reviewed your code. Loved your UI and animations. I have given some minor suggestion do check. Approving your merge

File: src/Pages/Home/Home.css (hunk 1 above)
Comment (zoef-yt) [reply src/Pages/Home/Home.css:115]: Done.

File: src/Pages/Components/Navbar/Navbar.jsx (hunk 2 above)
Comment (zoef-yt) [reply src/Pages/Components/Navbar/Navbar.jsx:31]: I think Github markdown computed your `<hr>` tag but I got your point. 

With `<hr>` tag.
> image

Without `<hr>` tag
> image

File: src/Pages/Components/Footer/Footer.jsx [hunk 3]
```diff
@@ -0,0 +1,11 @@
+import React from 'react';
+
+function Footer() {
```
Comment (muskanbagrecha) [inline src/Pages/Components/Footer/Footer.jsx:3]: Use arrow functions.

File: src/Pages/Components/Navbar/Navbar.css [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+	width: 100%;
+	display: flex;
+	justify-content: space-between;
+	align-items: center;
+}
+
+.navbar-brand {
+	margin: 0 2rem;
+	max-width: 100%;
+	width: 100%;
```
Comment (muskanbagrecha) [inline src/Pages/Components/Navbar/Navbar.css:11]: Suggestion: I don't think width is required here.

File: src/Pages/Components/Navbar/Navbar.css [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+
+.navbar-cta {
+	padding: 2rem;
+	display: flex;
+	justify-content: flex-end;
+	align-items: center;
+	gap: 2rem;
+
+	max-width: 100%;
+	width: 100%;
```
Comment (muskanbagrecha) [inline src/Pages/Components/Navbar/Navbar.css:33]: Can remove width here as well.

File: src/Pages/Components/Navbar/Navbar.jsx [hunk 6]
```diff
@@ -0,0 +1,41 @@
+import React from 'react';
```
Comment (muskanbagrecha) [inline src/Pages/Components/Navbar/Navbar.jsx:1]: import not required here.

File: src/Pages/Components/Navbar/Navbar.jsx [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+import { NavLink } from 'react-router-dom';
+import { HalfMoonIcon, SunIcon } from '../../../Assets/svg/AllSVG';
+import { useTheme } from '../../../Context';
+
+import './Navbar.css';
+function Navbar() {
+	const { theme, toggleTheme } = useTheme();
+	return (
+		<nav className='navbar-mobile'>
```
Comment (muskanbagrecha) [inline src/Pages/Components/Navbar/Navbar.jsx:10]: I think you can use a better name than navbar-mobile as it is ambiguous.

File: src/Pages/Components/Navbar/Navbar.jsx [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+					<h1>Game Plex</h1>
+				</NavLink>
+				<input type='search' className='text-field' placeholder='Search For Games' />
+				<div className='navbar-cta'>
+					<div className='flex-column' onClick={toggleTheme}>
+						{theme === 'dark' ? <SunIcon className='header-icon ' /> : <HalfMoonIcon className='header-icon' />}
+					</div>
+					<div className='profile-modal-holder'>
+						<div className='avatar avatar-sm '>ZS</div>
+						<div className='profile-modal'>
```
Comment (muskanbagrecha) [inline src/Pages/Components/Navbar/Navbar.jsx:22]: Extract the profile modal.

File: src/Pages/Home/Home.css [hunk 9]
```diff
@@ -0,0 +1,120 @@
...
+	position: absolute;
+	top: 10rem;
+	opacity: 0.5;
+	transition: all 0.3s ease;
+}
+
+.category-card:hover h3 {
+	top: 1rem;
+	opacity: 1;
+	transition: all 0.3s ease;
```
Comment (muskanbagrecha) [inline src/Pages/Home/Home.css:69]: You are using transition for every child so you can check if applying it to the parent will be sufficient as it gets too repetitive.

File: src/Pages/Home/Home.css [hunk 10]
```diff
@@ -0,0 +1,120 @@
...
+
+	.category-card h3 {
+		position: absolute;
+		top: 0;
+		width: 100%;
+		display: flex;
+		justify-content: center;
+		border-radius: 1rem;
+		opacity: 1;
+		z-index: 5;
```
Comment (muskanbagrecha) [inline src/Pages/Home/Home.css:107]: You could use variables for z-index.

File: src/Pages/Home/Home.jsx [hunk 11]
```diff
@@ -0,0 +1,66 @@
...
+						<h3>FPS</h3>
+						<img src='https://cdn1.epicgames.com/offer/cbd5b3d310a54b12bf3fe8c41994174f/EGS_VALORANT_RiotGames_S1_2560x1440-160cacc798ef1693cc0a339e869f1761'></img>
+						<button className='btn btn-primary'>Watch Now!</button>
+					</div>
+					<div className='category-card'>
+						<h3>Storyline</h3>
+						<img src='https://cdn1.epicgames.com/offer/cbd5b3d310a54b12bf3fe8c41994174f/EGS_VALORANT_RiotGames_S1_2560x1440-160cacc798ef1693cc0a339e869f1761'></img>
+						<button className='btn btn-primary'>Watch Now!</button>
+					</div>
+					<div className='category-card'>
```
Comment (muskanbagrecha) [inline src/Pages/Home/Home.jsx:40]: Extract this into a component.

File: src/index.css [hunk 12]
```diff
@@ -0,0 +1,42 @@
+@import url(https://unpkg.com/mockman-js@latest/dist/style.css);
+
+.game-plex-app {
+	display: grid;
+	grid-template-columns: 1fr;
+	grid-template-rows: 8rem 1fr 5rem;
+	grid-template-areas:
+		'navbar'
+		'main-content '
+		'footer  ';
```
Comment (muskanbagrecha) [inline src/index.css:10]: Remove the trailing spaces.

File: src/Pages/Home/Home.jsx [hunk 13]
```diff
@@ -0,0 +1,66 @@
...
+						<h3>Gameplay's</h3>
+						<img src='https://cdn1.epicgames.com/offer/cbd5b3d310a54b12bf3fe8c41994174f/EGS_VALORANT_RiotGames_S1_2560x1440-160cacc798ef1693cc0a339e869f1761'></img>
+						<button className='btn btn-primary'>Watch Now!</button>
+					</div>
+					<div className='category-card'>
+						<h3>Vlogs</h3>
+						<img src='https://cdn1.epicgames.com/offer/cbd5b3d310a54b12bf3fe8c41994174f/EGS_VALORANT_RiotGames_S1_2560x1440-160cacc798ef1693cc0a339e869f1761'></img>
+						<button className='btn btn-primary'>Watch Now!</button>
+					</div>
```
Comment (muskanbagrecha) [inline src/Pages/Home/Home.jsx:29]: Fetch the data using the api.

File: src/index.css [hunk 14]
```diff
@@ -0,0 +1,42 @@
...
+}
+
+@media screen and (max-width: 768px) {
+	.game-plex-app {
+		display: grid;
+		grid-template-columns: 1fr;
+		grid-template-rows: 14rem 1fr 5rem;
+		grid-template-areas:
+			'navbar'
+			'main-content '
```
Comment (muskanbagrecha) [inline src/index.css:35]: Remove trailing spaces.

**Review by muskanbagrecha — CHANGES_REQUESTED** [review]
Great work. Do look into the suggested changes.

File: src/Pages/Components/Footer/Footer.jsx (hunk 3 above)
Comment (zoef-yt) [reply src/Pages/Components/Footer/Footer.jsx:3]: done.

File: src/Pages/Components/Navbar/Navbar.css (hunk 4 above)
Comment (zoef-yt) [reply src/Pages/Components/Navbar/Navbar.css:11]: removed.

File: src/Pages/Components/Navbar/Navbar.css (hunk 5 above)
Comment (zoef-yt) [reply src/Pages/Components/Navbar/Navbar.css:33]: removed.

File: src/Pages/Components/Navbar/Navbar.jsx (hunk 6 above)
Comment (zoef-yt) [reply src/Pages/Components/Navbar/Navbar.jsx:1]: removed.

File: src/Pages/Components/Navbar/Navbar.jsx (hunk 7 above)
Comment (zoef-yt) [reply src/Pages/Components/Navbar/Navbar.jsx:10]: Will be changing this in future refractor

File: src/Pages/Components/Navbar/Navbar.jsx (hunk 8 above)
Comment (zoef-yt) [reply src/Pages/Components/Navbar/Navbar.jsx:22]: Done.

File: src/Pages/Home/Home.jsx (hunk 11 above)
Comment (zoef-yt) [reply src/Pages/Home/Home.jsx:40]: Done.

File: src/index.css (hunk 12 above)
Comment (zoef-yt) [reply src/index.css:10]: Done.

File: src/Pages/Home/Home.jsx (hunk 13 above)
Comment (zoef-yt) [reply src/Pages/Home/Home.jsx:29]: Done.

File: src/Pages/Home/Home.css (hunk 10 above)
Comment (zoef-yt) [reply src/Pages/Home/Home.css:107]: Will be doing it in PR #4

File: src/index.css (hunk 14 above)
Comment (zoef-yt) [reply src/index.css:35]: done.

File: src/Pages/Home/Home.css (hunk 9 above)
Comment (zoef-yt) [reply src/Pages/Home/Home.css:69]: I tried but it won't work like that.

**Comment by zoef-yt** [comment]
@muskanbagrecha did all the mentioned suggestions in pr #3

**Review by muskanbagrecha — APPROVED** [review]
Good to merge as you have implemented the suggested changes. Great animations and layout looks perfect on the listing page.

**Pull request merged**